app/views/application/_product_status_table.html.erb

```diff
@@ -42,7 +42,7 @@
       <th scope="col" class="c1-heading <%= 'hide-test' unless product.c2_test %>">QRDA-I</th>
       <th scope="col" class="c2-heading">QRDA-III</th>
       <th scope="col" class="c3-heading <%= 'hide-test' unless product.c1_test %>">Sample</th>
-      <th scope="col" class="c3-heading <%= 'hide-test' unless (product.c1_test and product.c2_test) or product.c3_test %>">QRDA-I</th>
```

Comment:
Going through this, i don't think this code needs to go into the model.  Instead

`<th scope="col" class="c1-heading">Sample</th>
      <th scope="col" class="c1-heading">QRDA-I</th>
      <th scope="col" class="c2-heading">QRDA-III</th>
      <th scope="col" class="c3-heading <%= 'hide-test' unless (product.c1_test and product.c3_test) %>">Sample</th>
      <th scope="col" class="c3-heading <%= 'hide-test' unless product.c3_test %>">QRDA-I</th>
      <th scope="col" class="c3-heading <%= 'hide-test' unless product.c3_test %>">QRDA-III</th>
      <th scope="col" class="c4-heading">QRDA-I</th>
      <th scope="col" class="c4-heading">QRDA-III</th>`

Reply:
So you think the logic for this whole section should be changes? I can definitely do that, but unless I'm not catching something, that seems to be the situation.

Reply:
Like I said, I'm not sure. Talk to @dczulada about it, he has the final say

Reply:
After a future look.  It only needs to be this.  The tests will already be "hidden" if the product is not a c3.  The only case that now needs a check is the Record Sample.  This is only included if C1 is true.

```
<th scope="col" class="c1-heading">Sample</th>
      <th scope="col" class="c1-heading">QRDA-I</th>
      <th scope="col" class="c2-heading">QRDA-III</th>
      <th scope="col" class="c3-heading <%= 'hide-test' unless product.c1_test">Sample</th>
      <th scope="col" class="c3-heading">QRDA-I</th>
      <th scope="col" class="c3-heading">QRDA-III</th>
      <th scope="col" class="c4-heading">QRDA-I</th>
      <th scope="col" class="c4-heading">QRDA-III</th>
```